wlc/main.py

```diff
@@ -337,16 +352,31 @@ def run(self):
 
 
 @register_command
-class ListComponents(Command):
+class ListComponents(ProjectCommand):
     """List components."""
 
     name = "list-components"
-    description = "Lists all components"
+    description = "Lists all components (optionally per project)"
 
     def run(self):
         """Main execution of the command."""
-        self.print(list(self.wlc.list_components()))
+        if self.args.object:
+            obj = self.get_object()
+
+            component_list = list(obj.list())
+            for component in component_list:
+                component.setattrvalue("project", obj)
+
+            self.print(component_list)
+        else:
+            self.print(list(self.wlc.list_components()))
 
+    def get_object(self):
+        """Return component object."""
+        obj = super(ProjectCommand, self).get_object()
```

Comment:
Looks wrong, probably should be `ListComponents` (or even better remove the args completely).